modules/internal/ChapelArray.chpl

```diff
@@ -1943,22 +1943,26 @@ module ChapelArray {
   }
 
   @unstable("casting an array to an array type is unstable due to being a new feature — please share any feedback you might have")
-  operator :(arr: [], type t: []) {
-    // Would like to write:
-    //   var a: t = [elem in arr] elem:t.eltType;
-    // but we don't support type queries on array types (yet)
-    // (see https://github.com/chapel-lang/chapel/issues/8543)
-    var a: t;
-
-    if !arr.isRectangular() || !a.isRectangular() {
+  operator :(rhs: [], type t: []) {
```

Comment:
@DanilaFe : Since you've been musing on arrays and runtime types lately, I thought you might enjoy seeing this extension to my first-draft cast operator which supports casting an array [of arrays of arrays of ...] to an array [of arrays of arrays] of the same overall structure, but with potentially different indices or values.  I was terrified that something would be complex or blow up due to the use of runtime types and recursive aspect of the implementation, but happily, it worked as expected.  Note that it's nicest to read in its non-diff form (but I don't think I can tag you in a comment as easily there).

Reply:
Thanks, Brad! (by the way, should that be `lhs` for `arr`? it is on the left of the cast, isn't it...)

Reply:
Hmm, good point.  I think I used rhs b/c I was thinking of it as `lhs = rhs.type` but that's fairly artificial (and I obviously didn't stick to that scheme with the result).  Maybe I should just call it `arg`?

Reply:
`arg` is fine, but personally I see nothing wrong with `lhs`.